src/native/corehost/hostfxr.h

```diff
@@ -363,4 +363,56 @@ typedef int32_t(HOSTFXR_CALLTYPE* hostfxr_get_dotnet_environment_info_fn)(
     hostfxr_get_dotnet_environment_info_result_fn result,
     void* result_context);
 
+struct hostfxr_framework_result
+{
+    size_t size;
+    const char_t* name;
+    const char_t* requested_version;
+    const char_t* resolved_version;
+    const char_t* resolved_path;
+};
+
+struct hostfxr_resolve_frameworks_result
+{
+    size_t size;
+
+    size_t resolved_count;
+    const struct hostfxr_framework_result* resolved_frameworks;
+
+    size_t unresolved_count;
+    const struct hostfxr_framework_result* unresolved_frameworks;
+};
+
+typedef void (HOSTFXR_CALLTYPE* hostfxr_resolve_frameworks_result_fn)(
+    const hostfxr_resolve_frameworks_result* result,
+    void* result_context);
+
+//
+// Resolves frameworks for a runtime config
+//
+// Parameters:
+//    runtime_config_path
+//      Path to the .runtimeconfig.json file
+//    parameters
+//      Optional. Additional parameters for initialization.
+//      If null or dotnet_root is null, the root corresponding to the running hostfx is used.
+//    callback
+//      Optional. Result callback invoked with result of the resolution.
+//      Structs and their elements are valid for the duration of the call.
+//    result_context
+//      Optional. Additional context passed to the result callback.
+//
+// Return value:
+//     0 on success, otherwise failure.
+//
+// String encoding:
+//     Windows  - UTF-16 (pal::char_t is 2-byte wchar_t)
+//     Unix     - UTF-8  (pal::char_t is 1-byte char)
```

Comment:
We've been inconsistent about this, but I've started to prefer "non-Windows" as opposed to "Unix". The only partially covers things like macOS, or FreeBSD.